### ISSUE TITLE
Java 14 SwitchExpression Check Support validation: CommentsIndentationCheck

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#8677]
Child of #8658 
Check documentation: https://checkstyle.sourceforge.io/config_misc.html#CommentsIndentation

Controls the indentation between comments and surrounding code. Comments are indented at the same level as the surrounding code. Detailed info about such convention can be found here

```
➜  src /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java                               
➜  src cat config.xml                                                                                                   
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="CommentsIndentation"/>
    </module>
</module>
➜  src cat TestClass.java                                                                                               
public class TestClass {
    private int a;
    private int b;

    public boolean equals1(Object obj) {
        int i = 1;
        switch (i) {
                    /* violation
                     *
                     */
            case 1:
                int count;
                a = a + b;
                b = a + a;
                count = b;
                    // violation
                break;
            case 2:
            case 3:
            case 4:
            case 5:
            case 6:
        }
        return false;
    }

    public boolean equals2(Object obj) {
        int i = 1;
        switch (i) {
                    /* violation
                     *
                     */
            case 1 -> {
                int count;
                a = a + b;
                b = a + a;
                count = b;
                    //violation
            }
        }
        return false;
    }
}

➜  src java -jar /home/nick/IdeaProjects/checkstyle/target/checkstyle-8.36-SNAPSHOT-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /home/nick/Desktop/enhanced-switch/src/TestClass.java:8:21: Block comment has incorrect indentation level 20, expected is 12, indentation should be the same level as line 11. [CommentsIndentation]
[ERROR] /home/nick/Desktop/enhanced-switch/src/TestClass.java:16:21: Comment has incorrect indentation level 20, expected is 16, indentation should be the same level as line 17. [CommentsIndentation]
[ERROR] /home/nick/Desktop/enhanced-switch/src/TestClass.java:30:21: Block comment has incorrect indentation level 20, expected is 8, 12, indentation should be the same level as line 29, 33. [CommentsIndentation]
[ERROR] /home/nick/Desktop/enhanced-switch/src/TestClass.java:38:21: Comment has incorrect indentation level 20, expected is 16, indentation should be the same level as line 37. [CommentsIndentation]
Audit done.
Checkstyle ends with 4 errors.

```

This check works as intended, so I think that we can close this one.

**Comment by pbludov** [comment]
Works as expected with comments between cases:
```java
public class TestClass {
    private void foo(int val) {
        switch(val) {
            // ok
            case 0:
        // violation
            case 1:
            // ok
            case 2: return;
                // violation
            default:
                // ok
                break;
        }
    }
    private boolean bar(int val) {
        return switch(val) {
            // ok
            case 1,
                 // ok
                 2,
            // violation
                 3,
                    // violation
                 4 -> true;
            // ok
            default -> false;
        };
    }
}
```
```
Starting audit...
[ERROR] TestClass.java:6:9: Comment has incorrect indentation level 8, expected is 12, 12, indentation should be the same level as line 5, 7. [CommentsIndentation]
[ERROR] TestClass.java:10:17: Comment has incorrect indentation level 16, expected is 12, 12, indentation should be the same level as line 9, 11. [CommentsIndentation]
[ERROR] TestClass.java:22:13: Comment has incorrect indentation level 12, expected is 17, indentation should be the same level as line 23. [CommentsIndentation]
[ERROR] TestClass.java:24:21: Comment has incorrect indentation level 20, expected is 17, indentation should be the same level as line 25. [CommentsIndentation]
Audit done.
```
